libs/qCC_db/ccIncludeGL.h

```diff
@@ -200,11 +200,14 @@ class ccGL
 			return false;
 		}
 
-		// Check if the point is in frustrum
-		if (Pp.x < -Pp.w  || Pp.y < -Pp.w || Pp.z < -Pp.w
-			|| Pp.x > Pp.w || Pp.y > Pp.w || Pp.z > Pp.w)
+		//Check if the point is in frustrum
+		if(checkInFrustrum)
 		{
-			return false;
+			if (Pp.x < -Pp.w  || Pp.y < -Pp.w || Pp.z < -Pp.w
```

Comment:
Yes, thanks it's more readable. My (maybe wrong) understanding of the thing here is that it's potentially less test the way it's done (vs. std::abs) because the condition checked by the `if` is true if only one of the condition is true (here we only have "or" opertators). So for example if `Pp.x < -Pp.w == true` it will return immediately => in this case it's only one test. with `std::abs(Pp.x) > Pp.w`, it first check the sign of Pp.x and then check the condition. I don't know if it's true and if it really matter.

Reply:
Well it's statistical. You have as much chance that the test stops at any of the sub-tests (in some cases it will quit right away, but in the same amount of cases it will stop after the 6th test - in average you'll make 3 tests). While in the second case you have at most 3 tests. So the worst case is definitely better, and in average you have 1.5 test. And std;;abs shouldn't be too long to apply (compared to a floating point test). So I believe its either the same or better?

Reply:
well I don't really know how is implemented std::abs: is it a square follow by a square root or is it a sign test (or is it something else)?
let's go for std::abs anyway.

Reply:
I hope it's not the first one! (that would be awfully slow :p). On Windows it's equivalent to fabs, and it's only an inversion of the 'sign' bit.

Reply:
yes but the sign is tested first, no?

Reply:
Nope, I believe the 'sign' bit is simply set to 0?

Reply:
yep sorry... it's the right solution then

Reply:
See http://aggregate.org/MAGIC/#Absolute%20Value%20of%20a%20Float